### DISCUSSION TITLE
enable SSL on AWS ec2 instance

### DISCUSSION THREAD

**Discussion by bigbean17** [ant-media/Ant-Media-Server#6276] [FAQs]
Hi,

I'm running the server on Amazon EC2 and trying to install ssl on the server. I followed this method
<img width="962" alt="image" src="https://github.com/ant-media/Ant-Media-Server/assets/69971426/d5820de6-028e-45ba-ae59-8ecc51a5eb05">
But it shows "The license key is invalid." after running  sudo `./enable_ssl.sh`. I'm not sure what caused this issue, I created the instance directly without changing from the marketplace image. Any help would be appreciated. Thanks~

**Comment by Mohit-3196** [comment]
Hi @bigbean17,
Thank you for reporting.
May I know which AMS version are you running from the AWS Marketplace?

Thank you,
Mohit

**Reply by bigbean17** [reply]
I believe its v282

**Reply by bigbean17** [reply]
I closed this discussion as we used cloudfront to get the certificate, but still would love to know why its not working, thanks~

**Reply by Mohit-3196** [reply]
Hi @bigbean17,
The free subdomain of antmedia.cloud is for users who have an Ant Media Server enterprise edition license. And because you are running it from the AWS marketplace that is why it is giving license errors for you.

Starting from Ant Media Server enterprise edition v2.9.0 the free subdomain will be available for the marketplace users as well. Version [2.9.0 has been released ](https://github.com/ant-media/Ant-Media-Server/releases) and it will be available on the marketplace by this weekend as well.

Meanwhile, if you share the Elastic IP address of your instance, I can create a subdomain of antmedia.cloud for you and share it with you.
Let me know if you need that and I can share my email with you so that you can share the elastic IP address.

Thank you.

**Comment by lexy10** [comment]
Hello there, I am finding it difficult to configure SSL for my Antmedia server.

Installed with docker on Ubuntu 24, I have also configured the AWS Application Load balancer for SSL and target groups. My domain routes fine to port :5080, but for port 5443, it throws "502 Bad Gateway". 

Please help. I've been battling this for days now and my license is reading...

**Reply by amarantmeida** [reply]
@lexy10 Please create a ticket with Ant Media Support and we will assist you further on your issue.

**Reply by lexy10** [reply]
@amarantmeida I was able to do it by logging into my EC2 instance through SSH and into the docker container and run the enable_ssl and it works fine through the port :5443 now.

**Reply by amarantmeida** [reply]
@lexy10 Glad to know that you got it working for you. Do let us know if you need any further assistance.

**Reply by lexy10** [reply]
@amarantmeida I think i should be able to add the command directly into Dockerfile since i already know the domain name and it's already registered right?

**Reply by amarantmeida** [reply]
@lexy10 Yes, you're correct. If you already know the domain name and have your SSL certificate and private key files, you can include the necessary commands directly in your Dockerfile to automate the SSL setup. This will allow the SSL configuration to be applied automatically whenever the Docker container is built or started.

**Reply by lexy10** [reply]
Alright, thank you for the clarification. if there's any issue, I'll let you know.
